File: FRONTEND/src/app/core/services/dashboard.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DashboardSummaryResponse, GroupedCountResponse } from '../../shared/models';

@Injectable({ providedIn: 'root' })
export class DashboardService {
  private readonly baseUrl = `${environment.apiBaseUrl}/dashboard`;

  constructor(private readonly http: HttpClient) {}

  getSummary(): Observable<DashboardSummaryResponse> {
    return this.http.get<DashboardSummaryResponse>(`${this.baseUrl}/summary`);
  }

  getAlarmsBySeverity(): Observable<GroupedCountResponse> {
    return this.http.get<GroupedCountResponse>(`${this.baseUrl}/alarms-by-severity`);
  }

  getIncidentsByStatus(): Observable<GroupedCountResponse> {
    return this.http.get<GroupedCountResponse>(`${this.baseUrl}/incidents-by-status`);
  }
}

File: Backend/dashboard-reporting-service/src/main/java/com/noc/dashboardservice/service/DashboardService.java
package com.noc.dashboardservice.service;

import com.noc.dashboardservice.dto.DashboardSummaryResponse;
import com.noc.dashboardservice.dto.GroupedCountResponse;

public interface DashboardService {

    DashboardSummaryResponse getSummary();

    GroupedCountResponse getAlarmsBySeverity();

    GroupedCountResponse getIncidentsByStatus();
}
